app/service/operation_svc.py

```diff
@@ -30,11 +30,14 @@ def __init__(self, data_svc, utility_svc, planner):
 
     async def run(self, op_id):
         self.log.debug('Starting operation: %s' % op_id)
+        await self.init_operation(op_id)
         operation = await self.data_svc.explode_operation(dict(id=op_id))
         try:
             for phase in operation[0]['adversary']['phases']:
                 self.log.debug('Operation phase %s: started' % phase)
-                await self.planner.execute(operation[0], phase)
+                cancel = await self.planner.execute(operation[0], phase, self)
+                if cancel == 'Cancel Requested':
```

Comment:
instead of 'cancel requested' can we use the 'cancelled' variable in the enum (in op_control)? so it would be OpState.CANCEL. this would be cleaner than a raw string here.

Reply:
That makes things a bit more finicky, but I think I've got it worked out.